source/loader/layers/validation/ur_valddi.cpp

```diff
@@ -9,7 +9,51 @@
  */
 #include "ur_validation_layer.hpp"
 
+#include <atomic>
+#include <functional>
+#include <map>
+#include <mutex>
+#include <shared_mutex>
+#include <utility>
+
 namespace validation_layer {
+struct {
+    std::map<void *, std::atomic<int>> counts;
```

Comment:
The reference count in UR is `uint32_t` so I think you should use a type that can hold at least UINT32_MAX references (`int64_t`)

Reply:
Done.